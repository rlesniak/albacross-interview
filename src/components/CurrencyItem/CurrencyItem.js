// @flow

import * as React from 'react';
import cx from 'classnames';
import { withHandlers, type HOC } from 'recompose';

import './CurrencyItem.scss';

type PropsType = {
  name: string,
  value: string,
  selected?: boolean,
  clickable?: boolean,
  onClick?: (name: string) => void
};

const enhance: HOC<*, PropsType> = withHandlers({
  handleOnClick: ({ name, onClick }) => () => {
    if (onClick) {
      onClick(name);
    }
  },
});

export const CurrencyItem = ({
  name, value, handleOnClick, selected, clickable,
}) => (
  <div
    className={cx('currency-item', {
      'currency-item--selected': selected,
      'currency-item--clickable': clickable,
    })}
    onClick={handleOnClick}
    title={value}
  >
    <b>{name}</b>: <br /> {value}
  </div>
);

export default enhance(CurrencyItem);
