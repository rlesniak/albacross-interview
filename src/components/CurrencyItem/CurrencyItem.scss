.currency-item {
  display: inline-block;
  vertical-align: top;
  margin: 5px 7px;
  text-align: center;
  width: 80px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  padding: 3px;

  &--selected {
    font-weight: bold;
    background: #e6f0f7;
    border-radius: 4px;
  }

  &--clickable {
    cursor: pointer;
  }
}