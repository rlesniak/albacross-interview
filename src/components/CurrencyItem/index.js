// @flow

export { default } from './CurrencyItem';
