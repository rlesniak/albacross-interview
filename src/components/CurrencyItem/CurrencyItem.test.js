import React from 'react';
import { shallow, mount } from 'enzyme';

import CurrencyItem, { CurrencyItem as PlainCurrencyItem } from './CurrencyItem';

describe('<PlainCurrencyItem />', () => {
  it('should render without crash', () => {
    const component = shallow(<PlainCurrencyItem name="USD" value="Dollar" handleOnClick={n => n} />);

    expect(component).toMatchSnapshot();
  });

  it('should have optional classes', () => {
    const component = shallow(<PlainCurrencyItem name="USD" value="Dollar" handleOnClick={n => n} selected clickable />);

    expect(component).toMatchSnapshot();
  });

  describe('<CurrencyItem />', () => {
    it('should call callback func', () => {
      const fn = jest.fn();
      const component = mount(<CurrencyItem name="USD" value="Dollar" onClick={fn} />);

      component.find('.currency-item').simulate('click');
      component.update();

      expect(fn).toHaveBeenCalled();
      expect(fn).toHaveBeenCalledWith('USD');
    });
  });
});

