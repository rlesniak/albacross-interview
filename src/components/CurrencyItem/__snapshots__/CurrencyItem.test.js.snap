// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlainCurrencyItem /> should have optional classes 1`] = `
<div
  className="currency-item currency-item--selected currency-item--clickable"
  onClick={[Function]}
  title="Dollar"
>
  <b>
    USD
  </b>
  : 
  <br />
   
  Dollar
</div>
`;

exports[`<PlainCurrencyItem /> should render without crash 1`] = `
<div
  className="currency-item"
  onClick={[Function]}
  title="Dollar"
>
  <b>
    USD
  </b>
  : 
  <br />
   
  Dollar
</div>
`;
